src/java/org/apache/cassandra/streaming/StreamingChannel.java

```diff
@@ -49,6 +49,14 @@ public static void unsafeSet(StreamingChannel.Factory factory)
         }
 
         StreamingChannel create(InetSocketAddress to, int messagingVersion, Kind kind) throws IOException;
+
+        default StreamingChannel create(InetSocketAddress to,
+                                        InetSocketAddress preferred,
+                                        int messagingVersion,
+                                        StreamingChannel.Kind kind) throws IOException
+        {
+            throw new UnsupportedOperationException();
```

Comment:
so fail if `BulkLoadConnectionFactory`?

Reply:
`BulkLoadConnectionFactory extends NettyStreamingConnectionFactory` right?

Reply:
Ah, but we still need to provide differentiated behavior in the `BulkLoadConnectionFactory` version or `create()`.

Reply:
Why provide the default interface method? Are there cases where you wouldn't implement this?

Reply:
Check the latest commit. There probably would have been an issue for the channel factory we create in `DirectStreamingConnectionFactory`.